src/challenge/views.py

```diff
@@ -289,6 +291,17 @@ def post(self, request):
             if challenge.first_blood is None:
                 challenge.first_blood = user
                 challenge.save()
+                hook = os.getenv("BLOOD_HOOK")
+                if hook:
+                    chall_clean = challenge.name.replace('`', '')
+                    team_clean = team.name.replace('`', '')
+                    body = {
+                             'content': "First blood on `" + chall_clean + "` by `" + team_clean + "`!",
+                             'allowed_mentions': {'parse': []},
+                             'username': 'First Bloods',
+                            }
+                    requests.post(hook, json=body)
```

Comment:
Completely missed this the first time I looked at this PR, this request isn't guaranteed to succeed. If this throws an exception, the challenge will still be updated, but the user and team will not be saved to the database leaving it in an inconsistent state, this will also cause any API client to see a generic 500 even though the flag was correct and was half scored. Wrapping this line in a try except avoids all of that. @ractf/backend this and emails is probably enough that we cant avoid adding celery to run background tasks.